### DISCUSSION TITLE
(many packages) Set ChocolateyPackageInstallLocation instead of printing out install location

### DISCUSSION THREAD

**Discussion by TheCakeIsNaOH** [chocolatey-community/chocolatey-packages#2254] [Ideas]
### Before starting, ensure that you have done the following.

- [X] I have verified that there is no other issue or discussion covering the feature or enhancement
- [X] I have verified there is no open pull request for the feature or enhancement
- [X] I have verified the package to improve is located in this repository

### What do you want to add?

There are quite a few packages that use `Get-AppInstallLocation` to find the installation location and then `Write-Host` the install location if found. Instead of printing out the install location, I suggest switching to setting `$env:ChocolateyPackageInstallLocation`, as that way, the software install location is hooked up to the Chocolatey CLI code for printing out the installation location, and for potential automatic uninstall improvements.

Here is an example of the current usage:
https://github.com/chocolatey-community/chocolatey-packages/blob/7c507b8ef5c022fdcdb7ca871f8653e5e80903f4/automatic/git.install/tools/chocolateyInstall.ps1#L29-L31

There appear to be about ~50 or so packages that use approximately this code.

### What problem will the feature/enhancement solve?

Currently, for packages that print out the result of `Get-AppInstallLocation`, they generally have redundant outputs that note the install location. 
For example, on my most recent `git.install` upgrade, the `chocolateyInstall.ps1` printed out this line:
`git.install installed to 'C:\Program Files\Git'`
And then close below that, Chocolatey printed out this line: 
`Software installed to 'C:\Program Files\Git\'`

### If the feature or enhancement is approved:

- [ ] I am willing to create a pull request to implement the feature or enhancement

### Additional Context

https://docs.chocolatey.org/en-us/faqs#the-install-location-displayed-is-incorrect-or-missing

**Comment by AdmiringWorm** [comment]
I have no problems with someone wanting to change this and set the `$env:ChocolateyPackageInstallLocation`.

@pauby and @gep13 what are your thoughts on this?

**Reply by pauby** [reply]
Does Chocolatey CLI allow that variable to be set and will it use it if it is?

**Reply by AdmiringWorm** [reply]
Nothing in Chocolatey CLI that prevents it, and it is the variable (at least one of the variables) used to determine the installation location outputted by Chocolatey CLI. I did have a look around, and I could not see anywhere we disallow setting this.

We even mention the possibility of setting this variable manually in the documentation here: https://docs.chocolatey.org/en-us/faqs#the-install-location-displayed-is-incorrect-or-missing

**Reply by AdmiringWorm** [reply]
One thing we should do if starting to set this variable, is to ensure that we never overwrite any existing values (unless in rare cases where it may be needed).

**Comment by TheCakeIsNaOH** [comment]
> One thing we should do if starting to set this variable, is to ensure that we never overwrite any existing values (unless in rare cases where it may be needed).

Would this be worth adding as a helper, or a feature to `Get-AppInstallLocation` in the core extension?

**Reply by AdmiringWorm** [reply]
I think that makes sense. Perhaps another helper in the `chocolatey-core.extension` package.

This new helper should probably by default have the logic of not overwriting the value if it is already existed, but with a `-Force` option to allow it being forced (for cases when the wrong location is set).

An extension to the existing cmdlet doesn't make sense to me.

Mind opening a issue over at <https://github.com/chocolatey-community/chocolatey-extensions> for this?

**Reply by TheCakeIsNaOH** [reply]
Done, agreed on the `-Force` behavior.

**Reply by AdmiringWorm** [reply]
Thanks for doing that.

To keep any comments/discussions over in that repository, I'll be closing this discussion now.

For anyone following this discussion, feel free to leave your thoughts over at https://github.com/chocolatey-community/chocolatey-extensions/issues/16
